OKR.md

```diff
@@ -0,0 +1,52 @@
+# Quarterly Objectives and Key Results
+
+We try to frame our ongoing work using a process based on quarterly Objectives and Key Results (OKRs). Objectives reflect outcomes that are challenging, but realistic. Results are tangible and measurable.
+
+## 2018 Q4
+
+**go-ipfs handles large datasets (1TB++) without a sweat**
+- `PX` - It takes less than 48 hours to transfer 1TB dataset over Fast Ethernet (100Mbps)
+- `PX` - It takes less than 12 hours to transfer 200GB sharded dataset over Fast Ethernet (100Mbps)
+- `PX` - There is a prototype implementation of GraphSync for go-ipfs
+- `PX` - There is a better and more performant datastore module (e.g Badger or better)
+- `P1` - Rewrite pinning data structures to support large data sets / many files performantly
+
+**The bandwidth usage is reduced significantly and is well kept under control**
+- `PX` - Users can opt out of providing every IPLD node (and only provide root hashes)
+- `PX` - "Bitswap improvements around reducing chattiness, decreasing bandwidth usage (fewer dupe blocks), and increasing throughput"
+
+**It is a joy to use go-ipfs programatically**
+- `PX` @magik6k - The Core API is finalized and released. Make it easier to import go-ipfs as a package
+- `PX` - go-ipfs-api exposes the new Core API
+- `PX` - go-ipfs Daemon uses the new Core API
+- `PX` - go-ipfs Gateway uses the new Core API
+- `PX` - go-ipfs-cms uses uses the new Core API
+- `PX` - The legacy non Core API is deprecated and the diagram on go-ipfs README is updated
+
+**go-ipfs becomes a well maintained project**
+- `PX` - The Go Contributing Guidelines are updated to contemplate expecations from Core Devs and instructions on how to be an effective contributor (e.g include PR templates)
+- `PX` - A Lead Maintainer Protocol equivalent is proposed, reviewed by the team, merged and implemented
+- `PX` - Every issue on https://waffle.io/ipfs/go-ipfs gets triaged (reviewed and labeled following https://github.com/ipfs/pm/blob/master/GOLANG_CORE_DEV_MGMT.md)
+- `PX` - Every package has tests and tests+code coverage are running on Jenkins
+- `PX` - There is an up-to-date Architecture Diagram of the Go implementation of IPFS that links packages to subsystems to workflows
+
+**gx becomes a beloved tool by the Go Core Contributors**
+- `PX` - 
+- `PX` - 
+
+**Complete outstanding endeavours and still high priorities from Q3**
+- `P0` @kevina - base32 is supported and enabled by default
+- `PX` - go-ipfs gets a unixfsV2 prototype
+- `PX` @djdv - IPFS Mount
```

Comment:
@djdv what would be the quantifiable KR for IPFS Mount?

Reply:
How does this sound?
>Add mutable methods to the new mount implementation and get it building+tested on all of our supported platforms

To recap the goal as it is now, we just want to have a new implementation of mount that takes advantage of all of the other new work that has been done (updated commands lib, CoreAPI, etc.), as well as add functionality such as a read+write MFS mountpoint. And we want to make sure this is actually true for all platforms we support (not just some of them).
More work will have to be done in our foundation itself (Golang) this quarter. Specifically, we need to assess the problems around dynamically linking to our fuse library on *nix, as we've done with Windows this quarter.

Reply:
@djdv is r+w mount across platforms seems like a lot for a quarter, but I don't know much about it or the current state. Is that an appropriately sized quarterly effort? Or is there something you're focusing on first that makes more sense for this goal?